### PR TITLE
fix: Reference Validation for Attendee-Ticket Relationship

### LINKED ISSUE #4659: No reference validation for attendee and ticket relationship
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-server

**Current behavior:**
Someone can add a reference to another event's ticket to an attendee on some other event

Event 155 has tickets 223 and 221

I can create an attendee with ticket 233 for event 255, where ticket 233 does not belong to the event 255

This causes serious referential integrity problems in mobile clients

**Expected behavior:**
Reference should be checked before persisting in database

**Steps to reproduce:**
```JSON
{
  "data": {
    "attributes": {
      "firstname": "Payer",
      "lastname": "Rich",
      "email": "payer.rich@example.com",
      "address": "address",
      "city": "example",
      "state": "example",
      "country": "IN",
      "is-checked_in": "false",
      "pdf-url": "http://example.com"
    },
    "type": "attendee",
    "relationships": {
      "event": { "data" : { "type" : "event", "id" : "{{event}}" } },
      "ticket": { "data" : { "type" : "ticket", "id" : "233" } }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by Masquerade0097**
Fixes #4659 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Before creating an Attendee it is checked whether the ticket_id associated with the request belongs to the event_id specified in the request. If it doesn't then Unprocessable Entity Exception is raised.

#### Changes proposed in this pull request:

-  Validate relationship between Ticket and Event before Attendee creation.

### CODE REVIEW

**Comment by Masquerade0097** [comment]
@iamareebjamal @shubham-padia Please review.

File: app/api/attendees.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def before_post(self, args, kwargs, data):
         """
         require_relationship(['ticket', 'event'], data)
 
+        ticket = db.session.query(Ticket).filter_by(id=data['ticket']).first()
```
Comment (iamareebjamal) [inline app/api/attendees.py]: `id=int(data['ticket'])` for extra safety

File: app/api/attendees.py [hunk 2]
```diff
@@ -31,6 +31,10 @@ def before_post(self, args, kwargs, data):
         """
         require_relationship(['ticket', 'event'], data)
 
+        ticket = db.session.query(Ticket).filter_by(id=int(data['ticket'])).first()
+        if ticket.event_id != int(data['event']):
```
Comment (iamareebjamal) [inline app/api/attendees.py]: Ticket can be `None`

Comment (Masquerade0097) [reply app/api/attendees.py]: Ok, I've added null check for `ticket` and both event and ticket relationships as well.

File: app/api/attendees.py [hunk 3]
```diff
@@ -31,6 +31,15 @@ def before_post(self, args, kwargs, data):
         """
         require_relationship(['ticket', 'event'], data)
 
+        ticket = db.session.query(Ticket).filter_by(id=int(data['ticket'])).first()
+        if ticket is not None:
+            if ticket.event_id != int(data['event']):
+                raise UnprocessableEntity({'pointer': '/data/relationships/ticket'},
+                                          "Ticket belongs to a different Event")
+        else:
+            raise UnprocessableEntity({'pointer': '/data/relationships/ticket'},
```
Comment (iamareebjamal) [inline app/api/attendees.py]: Can be flattened as instructed in previous review

**Comment by Masquerade0097** [comment]
@shubham-padia @niranjan94 Please review.

**Pull request merged**